lib/validate_repos.rb

```diff
@@ -0,0 +1,46 @@
+require "json"
+require "octokit"
+require "open-uri"
+require "yaml"
+
+class ValidateRepos
+  def initialize
+    Octokit.auto_paginate = true
+    @client = Octokit::Client.new(access_token: ENV.fetch("GITHUB_TOKEN"))
+  end
+
+  def github_repos_tagged_govuk
+    @github_repos_tagged_govuk ||= repos.map { |repo| repo["name"] }
+  end
+
+  def govuk_repo_names
+    @govuk_repo_names ||= JSON.parse(Net::HTTP.get_response(URI.parse("https://docs.publishing.service.gov.uk/repos.json")).body)
+      .map { |repo| repo["app_name"] }
+      .reject { |app| ignored_apps.include?(app) }
+  end
+
+  def untagged_repos
+    (govuk_repo_names - github_repos_tagged_govuk).join("\n")
+  end
+
+  def falsely_tagged_repos
+    (github_repos_tagged_govuk - govuk_repo_names).join("\n")
+  end
+
+  def repos
+    @client
+      .org_repos("alphagov", accept: "application/vnd.github.mercy-preview+json")
+      .select { |repo| repo.topics.to_a.include?("govuk") }
+      .reject(&:archived)
+      .reject { |repo| ignored_repos.include?(repo.full_name) }
```

Comment:
The [original FetchRepos class](https://github.com/alphagov/govuk-saas-config/blob/main/github/lib/fetch_repos.rb) was used by both repo tags check and a script that configured GitHub. The [exclusion of `ignored_repos` in the original code](https://github.com/alphagov/govuk-saas-config/blob/d80232700dbef861bd6c530cfd43b040873f0651/github/lib/fetch_repos.rb#L13) was used only by GitHub configuration script. The original govuk repo tags check didn't have ignore list.

We still want to be notified if `licensify` doesn't have `govuk` tag or if it's removed from the repos.json but still tagged.

I think we can ignore them **for now**, as there are some other considerations related to the ["Deploy out of sync" alerting](https://trello.com/c/VzDXayiM/3576-update-out-of-sync-deploy-alerting-to-account-for-multiple-apps-per-repo-3), which don't appear to be addressed.